internal/pkg/infrastructure/postgres/event.go

```diff
@@ -0,0 +1,255 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	pgClient "github.com/edgexfoundry/edgex-go/internal/pkg/db/postgres"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/errors"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v3/models"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/google/uuid"
+)
+
+const (
+	eventTableName = "\"core-data\".event"
```

Comment:
we need to discuss if we can change the schema to not contain dash

Reply:
Update schema name to `core_data`